code/modules/reagents/chemistry/reagents/medicine_reagents.dm

```diff
@@ -1757,6 +1757,43 @@
 				continue
 			movable_content.wash(clean_types)
 
+/datum/reagent/medicine/radaway
+	name = "RadAway"
+	description = "A potent but toxic chemical solution that binds with radioactive particles and render them inert. Applying this through spray or smoke will cleanse contaminanted surfaces."
+	color = "#9f5a2f"
+	var/old_insulation = RAD_NO_INSULATION
+	taste_description = "metallic dust"
+	self_consuming = TRUE
+
+/datum/reagent/medicine/radaway/reaction_mob(mob/living/M, method=TOUCH, reac_volume)
+	if(method == TOUCH || method == VAPOR)
+		M.wash(CLEAN_RAD) //you only get decontaminated if it's spray based, can't spam out 100 1u pills
+	
+/datum/reagent/medicine/radaway/on_mob_life(mob/living/carbon/M)
+	if(M.radiation > 0)
+		M.radiation -= min(M.radiation, 15)
+	M.adjustToxLoss(1*REM, 0)
+	..()
+
+/datum/reagent/medicine/radaway/on_mob_add(mob/living/L)
+	..()
+	//store the person's original insulation so they're only extra protected while it's in their system
+	old_insulation = L.rad_insulation
+	L.rad_insulation = RAD_LIGHT_INSULATION
+
+/datum/reagent/medicine/radaway/on_mob_end_metabolize(mob/living/L)
+	L.rad_insulation = old_insulation
+	if(iscarbon(L))
+		var/mob/living/carbon/C = L
+		C.vomit() //it binds with the radioactive particles inside you, and they have to come out somehow
```

Comment:
no
no no no
no no no no no
no
you don't want to see me use this
you don't want to see ANYONE use this
either vomit(stun = FALSE) or do something else

Reply:
added the stun = false clause